xds/src/main/java/io/grpc/xds/client/XdsClientImpl.java

```diff
@@ -141,12 +141,14 @@ public void handleResourceResponse(
   @Override
   public void handleStreamClosed(Status error) {
     syncContext.throwIfNotInThisSynchronizationContext();
-    cleanUpResourceTimers();
-    for (Map<String, ResourceSubscriber<? extends ResourceUpdate>> subscriberMap :
-        resourceSubscribers.values()) {
-      for (ResourceSubscriber<? extends ResourceUpdate> subscriber : subscriberMap.values()) {
-        if (!subscriber.hasResult()) {
-          subscriber.onError(error, null);
+    if (!error.isOk()) {
+      cleanUpResourceTimers();
```

Comment:
I believe that you always want to cleanup the timers.  The timers are to identify resources that don't exist and since this isn't connected to an xDS server you aren't really getting meaningful information in respect to resource existence.

Reply:
I noticed while I was making changes to tests. Done.